### ISSUE TITLE
Bundle configuration causes conflicting suggestions

### ISSUE THREAD

**Issue by Armaxis** [autonomousapps/dependency-analysis-android-gradle-plugin#701]
**Build scan link**
Will share in DM.

**Plugin version**
1.4.0

**Gradle version**
7.4.2

**Describe the bug**

TLDR: Bundle configuration causes conflicting suggestions when using exposing a class from a library's dependency, but not from the library itself.

**Reproducing**

I have a module `:common:gson:public` that exposes class `okio.ByteString`.

Dependency plugin has following bundle in its configuration
```
bundle('okio') {
  primary 'com.squareup.okio:okio'
  includeGroup 'com.squareup.okio'
}
```

Run a `reason` on the module `:common:gson:public`
```
> Task :common:gson:public:reason

----------------------------------------
You asked about the dependency 'com.squareup.okio:okio:3.0.0'.
You have been advised to add this dependency to 'api'.
----------------------------------------

Shortest path from :common:gson:public to com.squareup.okio:okio:3.0.0:
:common:gson:public
\--- com.squareup.wire:wire-gson-support:3.7.0
      \--- com.squareup.okio:okio:3.0.0

Source: debug, main
-------------------
(no usages)
```

So I add `okio` as `api` dependency and run `projectHealth`

```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':common:gson:public:projectHealth'.
> Unused dependencies which should be removed:
    api 'com.squareup.okio:okio:3.0.0'
  
  Transitively used dependencies that should be declared directly as indicated:
    api 'com.squareup.okio:okio:3.0.0' 
```

Run a `reason` again, here’s the output now
```
----------------------------------------
You asked about the dependency 'com.squareup.okio:okio:3.0.0'.
You have been advised to remove this dependency from 'api'.
----------------------------------------

Shortest path from :common:gson:public to com.squareup.okio:okio:3.0.0:
:common:gson:public
\--- com.squareup.wire:wire-gson-support:3.7.0
      \--- com.squareup.okio:okio:3.0.0

Source: debug, main
-------------------
(no usages)
```

All of this doesn’t make sense, like, how can it require and not require it at the same time, right?

But the mystery is solved if we remove the `bundle('okio')` configuration

Run `gw :common:gson:public:projectHealth` again

```
Execution failed for task ':common:gson:public:projectHealth'.
> Unused dependencies which should be removed:
    api 'com.squareup.okio:okio:3.0.0'
  
  Transitively used dependencies that should be declared directly as indicated:
    api 'com.squareup.okio:okio-jvm:3.0.0'
```

Now it makes sense! it’s a different dependency.
Check the reason `gw :common:gson:public:reason --id 'com.squareup.okio:okio-jvm:3.0.0'`
```
----------------------------------------
You asked about the dependency 'com.squareup.okio:okio-jvm:3.0.0'.
You have been advised to add this dependency to 'api'.
----------------------------------------

Shortest path from :common:gson:public to com.squareup.okio:okio-jvm:3.0.0:
:common:gson:public
\--- com.squareup.wire:wire-gson-support:3.7.0
      \--- com.squareup.okio:okio:3.0.0
            \--- com.squareup.okio:okio-jvm:3.0.0

Source: debug, main
-------------------
* Exposes class okio.ByteString (implies api).
```

So, the `:common:gson:public` doesn’t use classes from `com.squareup.okio:okio`, but uses classes from its dependency `com.squareup.okio:okio-jvm`
Dependency plugin should probably respect this kind of dependency and not complain about  `com.squareup.okio:okio`  because it is bundled with the `-jvm` dep

**Comment by autonomousapps** [comment]
Hm, I can't reproduce this in a test case. When I run reason on what I think is the same scenario, I get this output:

```
----------------------------------------
You asked about the dependency 'com.squareup.okio:okio:3.0.0'.
There is no advice regarding this dependency. It was removed because it matched a bundle rule.
----------------------------------------

Shortest path from :proj to com.squareup.okio:okio:3.0.0 for compileClasspath:
:proj
\--- com.squareup.okio:okio:3.0.0

Shortest path from :proj to com.squareup.okio:okio:3.0.0 for testCompileClasspath:
:proj
\--- com.squareup.okio:okio:3.0.0

Source: main
------------
(no usages)

Source: test
------------
(no usages)
```
Which makes sense, although it is probably confusing to the average reader. The project isn't using okio, but okio-jvm. And the advice is clean, as expected. Maybe the difference is that my test case is a kotlin-jvm library rather than an kotlin-android library?

Draft PR with not-failing test case at https://github.com/autonomousapps/dependency-analysis-android-gradle-plugin/pull/717

**Comment by autonomousapps** [comment]
I just reproduced it in register. It's happening because the bundle detection assumes that the "top-level" dependency is actually declared. But in this case, okio base (not -jvm) is just on the transitive classpath; it hasn't been declared.
